erpnext/manufacturing/doctype/production_plan/production_plan.py

```diff
@@ -568,7 +568,11 @@ def get_items_for_material_requests(doc, sales_order=None, company=None):
 					item_details = get_subitems(doc, data, item_details, bom_no, company,
 						include_non_stock_items, include_subcontracted_items, 1, planned_qty=planned_qty)
 		else:
-			item_master = frappe.get_doc('Item', data['item_code']).as_dict()
+			if data['item_code']:
```

Comment:
This won't fix the issue. You're still trying to access a key that doesn't exist. Use `data.get("item_code")` here instead.